src/Tree.h

```diff
@@ -129,6 +137,16 @@ Tree<T>::~Tree()
 }
 
 template <class T>
+void Tree<T>::setTemporary(bool state) {
+    m_isTemporary = state;
+}
+
+template <class T>
+bool Tree<T>::isTemporary() {
```

Comment:
Might be worth `const`ing the method - i.e.

    bool Tree<T>::isTemporary() const {